File: EcoChallengeBackend/EcoChallenge.Services/Database/Entities/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using EcoChallenge.Models.Enums;

namespace EcoChallenge.Services.Database.Entities
{
    public class Event
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey(nameof(Creator)), Column("creator_user_id")]
        public int CreatorUserId { get; set; }
        public virtual User? Creator { get; set; }
        [ForeignKey(nameof(Location)), Column("location_id")]
        public int LocationId { get; set; }
        public virtual Location? Location { get; set; }
        [Required, Column("title"), MaxLength(200)]
        public string? Title { get; set; }
        [Column("description")]
        public string? Description { get; set; }
        [Required, ForeignKey(nameof(EventType)), Column("event_type_id")]
        public int EventTypeId { get; set; }
        public virtual EventType? EventType { get; set; }
        [Column("max_participants")]
        public int MaxParticipants { get; set; } = 0;
        [Column("current_participants")]
        public int CurrentParticipants { get; set; } = 0;

        [Required, Column("event_date"), DataType(DataType.Date)]
        public DateTime EventDate { get; set; }
        [Required, Column("event_time"), DataType(DataType.Time)]
        public TimeSpan EventTime { get; set; }
        [Column("duration_minutes")]
        public int DurationMinutes { get; set; } = 120;
        [Column("equipment_provided")]
        public bool EquipmentProvided { get; set; } = false;
        [Column("equipment_list")]
        public string? EquipmentList { get; set; }
        [Column("meeting_point")]
        public string? MeetingPoint { get; set; }

        [Required, ForeignKey(nameof(Status)), Column("status_id")]
        public int StatusId { get; set; }
        public virtual EventStatus? Status { get; set; }
        [Column("admin_approved")]
        public bool AdminApproved { get; set; } = false;
        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual ICollection<Photo>? Photos { get; set; }
        public virtual ICollection<EventParticipant>? Participants { get; set; }
        public virtual ICollection<GalleryShowcase>? GalleryShowcases { get; set; }
        public virtual ICollection<ActivityHistory>? ActivityLogs { get; set; }
    }
}

File: EcoChallengeFrontend/ecochallenge_mobile/lib/models/event.dart
class Event {
  final int id;
  final int creatorUserId;
  final int locationId;
  final String? title;
  final String? description;
  final List<String>? photoUrls;
  final int eventTypeId;
  final int maxParticipants;
  final int currentParticipants;
  final DateTime eventDate;
  final String eventTime;
  final int durationMinutes;
  final bool equipmentProvided;
  final String? equipmentList;
  final String? meetingPoint;
  final int statusId;
  final bool isPaidRequest;
  final int? relatedRequestId;
  final bool adminApproved;
  final DateTime createdAt;
  final DateTime updatedAt;

  Event({
    required this.id,
    required this.creatorUserId,
    required this.locationId,
    this.title,
    this.description,
    this.photoUrls,
    required this.eventTypeId,
    required this.maxParticipants,
    required this.currentParticipants,
    required this.eventDate,
    required this.eventTime,
    required this.durationMinutes,
    required this.equipmentProvided,
    this.equipmentList,
    this.meetingPoint,
    required this.statusId,
    required this.isPaidRequest,
    this.relatedRequestId,
    required this.adminApproved,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      creatorUserId: json['creatorUserId'],
      locationId: json['locationId'],
      title: json['title'],
      description: json['description'],
      photoUrls: (json['photoUrls'] as List?)?.map((e) => e.toString()).toList(),
      eventTypeId: json['eventTypeId'],
      maxParticipants: json['maxParticipants'],
      currentParticipants: json['currentParticipants'],
      eventDate: DateTime.parse(json['eventDate']),
      eventTime: json['eventTime'] != null
          ? _formatTimeSpan(json['eventTime'])
          : '00:00',
      durationMinutes: json['durationMinutes'],
      equipmentProvided: json['equipmentProvided'],
      equipmentList: json['equipmentList'],
      meetingPoint: json['meetingPoint'],
      statusId: json['statusId'],
      isPaidRequest: json['isPaidRequest'],
      relatedRequestId: json['relatedRequestId'],
      adminApproved: json['adminApproved'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  static String _formatTimeSpan(dynamic timeSpanJson) {
    final h = timeSpanJson['hours'] ?? 0;
    final m = timeSpanJson['minutes'] ?? 0;
    return '${h.toString().padLeft(2, '0')}:${m.toString().padLeft(2, '0')}';
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'creatorUserId': creatorUserId,
      'locationId': locationId,
      'title': title,
      'description': description,
      'photoUrls': photoUrls,
      'eventTypeId': eventTypeId,
      'maxParticipants': maxParticipants,
      'currentParticipants': currentParticipants,
      'eventDate': eventDate.toIso8601String(),
      'eventTime': eventTime,
      'durationMinutes': durationMinutes,
      'equipmentProvided': equipmentProvided,
      'equipmentList': equipmentList,
      'meetingPoint': meetingPoint,
      'statusId': statusId,
      'isPaidRequest': isPaidRequest,
      'relatedRequestId': relatedRequestId,
      'adminApproved': adminApproved,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
